wdl-lint/src/rules/blank_lines_between_elements.rs

```diff
@@ -0,0 +1,566 @@
+use wdl_ast::SupportedVersion;
+use wdl_ast::SyntaxKind;
+use wdl_ast::SyntaxNode;
+use wdl_ast::SyntaxToken;
+use wdl_ast::ToSpan;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the blanks between elements rule.
+const ID: &str = "BlankLinesBetweenElements";
+
+/// Creates an excessive blank line diagnostic.
+fn excess_blank_line(span: Span) -> Diagnostic {
+    Diagnostic::note("extra blank line(s) found")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("remove the blank line(s)")
+}
+
+/// Creates a missing blank line diagnostic.
+fn missing_blank_line(span: Span) -> Diagnostic {
+    Diagnostic::note("missing blank line")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("add a blank line before this element")
+}
+
+/// Track the position within a document
+#[derive(Debug, Default, Clone, Copy, PartialEq)]
+enum State {
+    /// Outside tracked sections
+    #[default]
+    Outside,
+    /// In a `Input` section
+    InputSection,
+    /// In a `Output` section
+    OutputSection,
+    /// In a `Meta` section
+    MetaSection,
+    /// In a `Parameter Meta` section
+    ParameterMetaSection,
+    /// In a `Runtime` section
+    RuntimeSection,
+}
+
+/// Detects unsorted input declarations.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct BlankLinesBetweenElementsRule {
+    /// Store whether we are in certain blocks
+    state: State,
+}
+
+impl Rule for BlankLinesBetweenElementsRule {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Ensures that WDL elements are spaced appropriately."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "There should be a blank line between each WDL element at the root indentation level (such \
+         as the import block and any task/workflow definitions) and between sections of a WDL task \
+         or workflow. Never have a blank line when indentation levels are changing (such as \
+         between the opening of a workflow definition and the meta section). There should also \
+         never be blanks within a `meta`, `parameter_meta`, `input`, `output`, `runtime`, \
+         `requirements`, or `hints` section. For workflows, the `workflow body` includes any \
+         private declarations, call statements, conditional statements, and scatter statements. \
+         Within a workflow body, individual elements may optionally be separated by a blank line. \
+         See example (https://github.com/stjude-rust-labs/wdl/discussions/11) for a complete WDL \
+         document with proper spacing between elements. Note the blank lines between `meta`, \
+         `parameter_meta`, `input`, workflow body / `command` section, `output`, and `runtime` \
+         for the example task. The blank line between the workflow definition and the task \
+         definition is also important."
```

Comment:
```suggestion
         "
```
I don't think this block of text really belongs. The link doesn't go directly to the example making it an awkward inclusion IMO. We do want examples (for most if not all of our rules), but I don't think this is the right approach for that. A separate PR later can deal with adding examples in a more robust way.